doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -510,6 +510,19 @@ Requests to the environment
 
 .. seealso:: Section :ref:`locating-notations`
 
+.. _printing-flags:
+
+Printing flags
+-------------------------------
+
+.. flag:: Fast Name Printing
+
+   When turned on, |Coq| uses an asymptotically faster algorithm for the
+   attribution of non-ambiguous names of bound variables while printing terms.
```

Comment:
"attribution of non-ambiguous names of bound variables" is a mouthful.  Is there a simpler way to say this?  Is "attribution" the best word--not something like "naming"?  I have trouble guessing what "attribution" means here.  (Caveat: still lots of things I don't know about Coq.)

It would be helpful to the reader to give a short concrete example, for example: "ABC" will be printed as "DEF" when the flag is enabled, but otherwise would be printed as "GHI".  This could also illustrate what you mean by "less clever" and "less elegant" which are vague terms.  Is the less-good appearance perhaps related to generating more bound variable names rather than reusing them?

Reply:
It's not quite naming, as the name hint is already there. It's mostly about desambiguation.

Regarding cleverness, I don't have a short insightful example, but indeed in general more names are generated, e.g. if a constant name clashes in one subterm, the slow algorithm will forbid this name only in the subterm while it will be globally forbidden with the fast one.

Reply:
Maybe add "It will generate more names rather than reusing certain names across subterms."  That explains less clever/elegant a bit without going into detail.

And maybe "generation" would be clearer than "attribution".